### PR TITLE
Add support for span<T> as a sequence view-type

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds support for std::span as a sequence view-type in Ice C++.

It also updates the mapping for view-type from const T& to T, i.e. view types are always passed by value (not const reference). And for consistency, "array" parameters are now also passed by value, e.g. we now transmit a byte array as `pair<byte, byte>` instead of `const pair<byte, byte>&`.

See https://doc.zeroc.com/ice/3.7/language-mappings/c++11-mapping/customizing-the-c++11-mapping/the-cpp-type-and-cpp-view-type-metadata-directives-with-c++11#id-.Thecpp:typeandcpp:viewtypeMetadataDirectiveswithC++11v3.7-Avoidingcopieswithcpp:view-type

Note that in Ice 3.8, view-type no longer applies when unmarshaling, while array still does.

### CODE REVIEW

File: cpp/include/Ice/Connection.h [hunk 1]
```diff
@@ -9,7 +9,6 @@
 #include "ProxyF.h"
 #include "ValueF.h"
 #include "Exception.h"
-#include "StreamHelpers.h"
```
Comment (bernardnormier) [inline cpp/include/Ice/Connection.h:12]: Unrelated cleanup - should have been included in a separate PR.

File: cpp/include/Ice/StreamHelpers.h [hunk 2]
```diff
@@ -13,6 +13,10 @@
 #include <optional>
 #include <string_view>
 
+#if __has_include(<span>)
```
Comment (bernardnormier) [inline cpp/include/Ice/StreamHelpers.h:16]: That's the preferred and portable way to conditionally include span. Span is a c++20 and up feature.

File: cpp/src/slice2cpp/CPlusPlusUtil.cpp [hunk 3]
```diff
@@ -94,6 +94,51 @@ namespace
         }
     }
 
+    // Do we pass this type by value when it's an input parameter?
+    bool inputParamByValue(const TypePtr& type, const StringList& metaData)
```
[the remark this answers is not in the document]
Comment (pepone) [reply cpp/src/slice2cpp/CPlusPlusUtil.cpp:98]: Sounds fine, assuming the most common is to use the shared_ptr and not store it.

**Review by pepone — APPROVED** [review]
Looks good
